js/src/Builder.js

```diff
@@ -626,8 +717,31 @@ function _update_data (update_model, update_map, kind, should_draw) {
 
   // metabolite data
   if (update_metabolite_data && update_map && this.map !== null) {
-    met_data_object = data_styles.import_and_check(this.options.metabolite_data,
-                                                   'metabolite_data')
+
+    if (difference_mode_active && this.options.metabolite_data !== null) {
+
+      var difference_metabolite_data = [this.options.metabolite_data[reference], this.options.metabolite_data[target]]
+      met_data_object = data_styles.import_and_check(difference_metabolite_data, 'metabolite_data')
+    } else {
+      if (this.options.metabolite_data !== null) {
+        met_data_object = data_styles.import_and_check(this.options.metabolite_data[reference], 'metabolite_data')
```

Comment:
I'm a little confused with this one. I think we need to change the unit test to expect the return values in the new way? It was 'var data = []' now it is  'var data = [[],[]] // [array of names][array of data]'